src/code/init.js

```diff
@@ -138,6 +138,10 @@ window.plugin.wasabee.init = () => {
     // load Wasabee-Defense keys if logged in
     window.map.fire("wasabeeDkeys", { reason: "startup" }, false);
   }
+
+  window.map.on("wdialog", (dialog) => {
+    postToFirebase({ id: "analytics", action: dialog.constructor.TYPE });
+  });
```

Comment:
I like this, it makes it fully async so the UI doesn't lag on it.

Reply:
is it really async ? I used this to avoid a circular dep

Reply:
my understanding is that events are processed at a different phase of the (main) event loop. so the firing doesn't just cause execution to jump to the handler, rather it creates an entry in the queue which will be processed when the loop gets to the queue.

https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Overview_of_Events_and_Handlers